src/main/java/lotto/domain/LottoMachine.java

```diff
@@ -0,0 +1,46 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LottoMachine {
+    public static final int LOTTO_UNIT_PRICE = 1000;
+    private static final int MAX_PURCHASE_AMOUNT = 100000;
+
+    private final LottoNumberGenerator lottoNumberGenerator;
+
+    public LottoMachine(final LottoNumberGenerator lottoNumberGenerator) {
+        this.lottoNumberGenerator = lottoNumberGenerator;
+    }
+
+    public List<Lotto> purchase(final int purchaseAmount) {
+        validatePurchaseAmount(purchaseAmount);
+        int lottoAmount = purchaseAmount / LOTTO_UNIT_PRICE;
+        List<Lotto> lottos = new ArrayList<>();
+        for (int i = 0; i < lottoAmount; i++) {
+            Lotto lotto = createLotto();
+            lottos.add(lotto);
+        }
+        return lottos;
```

Comment:
아래와 같이 변경해볼 수도 있을 것 같아요. 😉
```suggestion
        return Stream.generate(this::createLotto)
            .limit(lottoAmount)
            .toList();
```

Reply:
아직 Stream을 사용하는 것이 익숙하지 않았었는데, 의식적으로 사용할 수 있도록 연습을 해봐야겠어요.
감사합니다!